langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/statements/foreach/ForeachIterableObjectTest.java

```diff
@@ -84,33 +84,55 @@ public void testNestedIterableObject() {
 
     }
 
+    @Test
+    public void testIterableObjectReturnedByRangeExpression() {
+        BValue[] returns = BRunUtil.invoke(program, "testIterableObjectReturnedByRangeExpression");
+
+        BValueArray arr = (BValueArray) returns[0];
+        Assert.assertEquals(arr.size(), 3);
+        int i = 0;
+        Assert.assertEquals(arr.getInt(i++), 1);
+        Assert.assertEquals(arr.getInt(i++), 2);
+        Assert.assertEquals(arr.getInt(i++), 3);
+    }
+
+    @Test
+    public void testIterableDistinctObjectReturnedByRangeExpression() {
+        BValue[] returns = BRunUtil.invoke(program, "testIterableDistinctObjectReturnedByRangeExpression");
+
+        BValueArray arr = (BValueArray) returns[0];
+        Assert.assertEquals(arr.size(), 3);
+        int i = 0;
+        Assert.assertEquals(arr.getInt(i++), 1);
+        Assert.assertEquals(arr.getInt(i++), 2);
+        Assert.assertEquals(arr.getInt(i++), 3);
```

Comment:
Shall we do the assertions in ballerina itself?

Reply:
Done
3ad22b602be3ec6627c2fb16664d805e50e39b57